test/tools/ossfuzz/protomutators/YulProtoMutator.cpp

```diff
@@ -0,0 +1,2081 @@
+#include <test/tools/ossfuzz/protomutators/YulProtoMutator.h>
+
+#include <libyul/Exceptions.h>
+
+#include <src/text_format.h>
+
+using namespace solidity::yul::test::yul_fuzzer;
+
+using namespace protobuf_mutator;
+
+using namespace std;
+
+template <typename Proto>
+using YPR = YulProtoCBRegistration<Proto>;
+using YPM = YulProtoMutator;
+using PrintChanges = YPM::PrintChanges;
+
+MutationInfo::MutationInfo(ProtobufMessage* _message, string const& _info):
+	ScopeGuard([&]{ exitInfo(); }), m_protobufMsg(_message)
+{
+	print("----------------------------------");
+	print("YULMUTATOR: " + _info);
+	print("Before");
+	print(SaveMessageAsText(*m_protobufMsg));
+
+}
+
+void MutationInfo::exitInfo()
+{
+	print("After");
+	print(SaveMessageAsText(*m_protobufMsg));
+}
+
+template <typename T>
+void YulProtoMutator::functionWrapper(
+	CustomFuzzMutator<T> const& _callback,
+	T* _message,
+	unsigned int _seed,
+	unsigned _period,
+	string const& _info,
+	PrintChanges _printChanges)
+{
+	YulRandomNumGenerator random(_seed);
+
+	if (_seed % _period == 0)
+	{
+		if (_printChanges == PrintChanges::Yes)
+		{
+			MutationInfo m{_message, _info};
+			_callback(_message, random);
+		}
+		else
+			_callback(_message, random);
+	}
+}
+
+// Add assignment to m/s/calldataload(0)
+static YPR<AssignmentStatement> assignLoadZero(
+	[](AssignmentStatement* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<AssignmentStatement>(
+			[](AssignmentStatement* _message, YulRandomNumGenerator& _rand)
+			{
+				_message->clear_expr();
+				_message->set_allocated_expr(YPM::loadFromZero(_rand));
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Assign load from zero"
+		);
+	}
+);
+
+// Invert condition of an if statement
+static YPR<IfStmt> invertIfCondition(
+	[](IfStmt* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<IfStmt>(
+			[](IfStmt* _message, YulRandomNumGenerator&)
+			{
+				if (_message->has_cond())
+				{
+					auto notOp = new UnaryOp();
```

Comment:
Are raw pointers being used throughout the fuzzer? If not, maybe use `unique_ptr` here?